pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -373,6 +405,89 @@ func (r *SystemReconciler) systemAppDeploymentResourceMutator(desired, existing
 	return update, nil
 }
 
+func (r *SystemReconciler) systemZyncEnvVarMutator(desired, existing *k8sappsv1.Deployment) (bool, error) {
+	update := false
+
+	// Reconcile ZYNC_AUTHENTICATION_TOKEN env var
+	for idx := range existing.Spec.Template.Spec.Containers {
+		tmpChanged := helper.EnvVarReconciler(
+			desired.Spec.Template.Spec.Containers[idx].Env,
+			&existing.Spec.Template.Spec.Containers[idx].Env,
+			"ZYNC_AUTHENTICATION_TOKEN")
+		update = update || tmpChanged
+	}
+
+	return update, nil
+}
+
+func (r *SystemReconciler) routesExist() (bool, error) {
+	expectedRoutes := 6
+	opts := k8sclient.ListOptions{
+		Namespace: r.apiManager.GetNamespace(),
+	}
+
+	routes := routev1.RouteList{}
+	err := r.Client().List(r.Context(), &routes, &opts)
+	if err != nil {
+		return false, err
+	}
+
+	if len(routes.Items) < expectedRoutes {
+		r.Logger().Info(fmt.Sprintf("Missing routes - expected at least %d routes but only found %d", expectedRoutes, len(routes.Items)))
+		return false, nil
+	}
+	return true, nil
+}
+
+func (r *SystemReconciler) isZyncReady() (bool, error) {
```

Comment:
this feels to me like a function that could be used by other reconcilers in future as well? Possibly a shared func? Not a big deal if it stays

Reply:
I'll move this function as well